src/gpu/intel/ocl/micro_sdpa.cl

```diff
@@ -211,8 +211,12 @@ DECLARE_2D_TILE_RSELECT(a_scale_tile_type, SUBGROUP_SIZE, ugemm_vs_sg_tile_n, 1,
 __attribute__((intel_reqd_sub_group_size(SUBGROUP_SIZE))) kernel void
 micro_sdpa(const global KEY_DATA_T *K, const global QRY_DATA_T *Q,
         const global VAL_DATA_T *V, global DST_DATA_T *A,
-        const global SCALE_DATA_T *scale_ptr, int d, int k, int q,
-        const global KEY_ATTR_SCALES_DATA_T *K_scales,
+#if HOST_SIDE_SCALE
+        const float scale, const float iscale,
+#else
+        const global SCALE_DATA_T *scale_ptr,
+#endif
```

Comment:
It feels like i(nverted)scale can be computed down below to have the same number of arguments.
Once one for both cases, I'd suggest to introduce `SCALE_T` (or something along the way) that will be `float` or `global SCALE_DATA_T *` depending on whether it's a host side scale or not.

Macros in kernel signature is hard to deal with at caller and callee levels, better avoid it whenever possible.

Reply:
See my attempt at commit: https://github.com/uxlfoundation/oneDNN/commit/fd3c5eb5c21a7102080ad017f35a9096b5c51c63 . So, this is one argument, but to me this obfuscates things and puts a division (to get iscale) back into the kernel. @umar456 and I went back and forth on this and pulled all compute back to the host. Less work device side. Also, `WITH_ATTN_MASK` is a macro in the kernel signature already, so the mess may have already been made. Lemme know!

Reply:
If the scale is host scalar, it can be actually passed as pointer, thus, no need to have a different argument signature. And it can be passed already inverted so the kernel can always multiply on it unconditionally, this information is available at execution time and access to the scale is identical to the access from the kernel.

Reply:
Discussed offline.
Due to the CPU nature of the host scalar, it can't be passed inside the kernel as a pointer, only as a plain float value.
If the final decision is to go with two values, agreed on putting a comment to explain the benefits of it. Otherwise, the scale handling would be self-documented in the code.